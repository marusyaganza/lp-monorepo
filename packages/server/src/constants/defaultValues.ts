import { Language, Role, SortWordsBy, Tense } from '../generated/graphql';

export const WORDS_PER_PAGE = 5;
export const DEFAULT_LANGUAGE = Language.English;
export const DEFAULT_WORDS_SORT_BY = SortWordsBy.UpdatedAt;
export const DEFAULT_ROLE = Role.Member;
export const DEFAULT_TENSE = Tense.Pind;
export const DEFAULT_TOKEN_TTL = 7;
// to clean db every hour + 10 min just in case
export const DEMO_DB_TTL = 4200;
export const DEFAULT_MONGO_CONNECTION = 'mongodb://localhost:27017';
export const DEFAULT_OPTIONS__SAMPLE_SIZE = 50;
