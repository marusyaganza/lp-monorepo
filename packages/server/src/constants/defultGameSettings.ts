import { Game, Language, SettingsByGame } from '../generated/graphql';

export const DEFAULT_GAMES_SETTINGS: SettingsByGame = {
  [Game.Audio]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Image]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Speaking]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.TypeWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectDef]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Conjugation]: {
    wordsPerGame: 2,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  },
  [Game.Gender]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  }
};
