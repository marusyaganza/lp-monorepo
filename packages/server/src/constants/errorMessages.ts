export const ERROR_MESSAGES = {
  NOT_AUTHENTICATED: 'Login to perform this operation',
  NOT_AUTHORIZED: 'You do not have permission to perform this operation',
  INCORRECT_CREDS: 'email or password is incorrect',
  SIGN_UP_FAILED: 'sign up operation failed',
  SING_UP_UNAVAILABLE:
    'Sign up operation is not available now. Sorry for inconvenience.',
  GAME_NOT_FOUND: 'Game not found',
  GAME_GENERATION_FAILED: 'Game data generation failed'
};
