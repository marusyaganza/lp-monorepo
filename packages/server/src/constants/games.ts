import { Game, GameConfig, Language } from '../generated/graphql';
export const GAMES: GameConfig[] = [
  {
    name: 'WordEcho',
    id: 'audio',
    desc: 'Listen, Type, Master. Sharpen language skills through auditory engagement.',
    type: Game.Audio,
    orderNum: 0,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'LexiSense',
    desc: 'Define & Conquer.Match words with their meanings intuitively.',
    id: 'select_definition',
    type: Game.SelectDef,
    orderNum: 1,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'Definify: Match & Learn',
    desc: 'Associate definitions with the right words. A captivating word puzzle adventure!',
    id: 'select_word',
    type: Game.SelectWord,
    orderNum: 2,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'WordRecall: Define & Type',
    desc: 'Type words matching given definitions. Boost vocabulary in engaging rounds!',
    id: 'type_word',
    type: Game.TypeWord,
    orderNum: 3,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'PictureWord Blitz',
    desc: 'Identify objects from pictures and type their names to enhance vocabulary skills!',
    id: 'image',
    type: Game.Image,
    orderNum: 4,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'VoiceWord Challenge',
    desc: 'Speak words from clues and images. Sharpen your vocabulary and speaking skills fast!',
    id: 'speaking',
    type: Game.Speaking,
    orderNum: 5,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'Conjugation Quest',
    desc: 'Master verb conjugations across tenses and languages in fast-paced, interactive challenges!',
    id: 'conjugation',
    type: Game.Conjugation,
    orderNum: 6,
    languages: [Language.Spanish]
  },
  {
    name: 'GenderWord Challenge',
    desc: 'Choose nouns and match their gender. Test linguistic skills in thrilling rounds!',
    id: 'gender',
    type: Game.Gender,
    orderNum: 7,
    languages: [Language.Spanish]
  }
];
