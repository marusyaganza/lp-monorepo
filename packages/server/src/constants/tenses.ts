import { Tense } from '../generated/graphql';

// Uncomment more tenses as needed
export const TENSES: Record<Tense, string> = {
  pind: 'Presente indicativo',
  pprf: 'Pretérito perfecto simple',
  impf: 'Imperativo',
  pret: 'Pretérito imperfecto',
  futr: 'Futuro indicativo',
  cond: 'Condicional',
  psub: 'Presente subjentivo'
  // gppt: 'Gerundio'
  //   pisb1: 'Pretérito imperfecto',
  //   pisb2: 'Pretérito imperfecto 2',
  //   fsub: 'Futuro',
  //   ppci: 'Pretérito perfecto compuesto',
  //   ppsi: 'Pretérito pluscuamperfecto',
  //   pant: 'Pretérito anterior',
  //   fpin: 'Futuro perfecto',
  //   cpef: 'Condicional perfecto',
  //   ppfs: 'Pretérito perfecto',
  //   ppss1: 'Pretérito pluscuamperfecto',
  //   ppss2: 'Pretérito pluscuamperfecto 2',
  //   fpsb: 'Futuro perfecto',
};
