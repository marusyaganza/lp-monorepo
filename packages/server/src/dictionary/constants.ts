import { Tense } from 'generated/graphql';

export type TagType = {
  tag: string;
  replacement: string;
};

export type ComplexTagType = {
  opening: string;
  closing: string;
};

export const TAGS: TagType[] = [
  { tag: '{b}', replacement: '<b>' },
  { tag: '{/b}', replacement: '</b>' },
  { tag: '{ldquo}', replacement: '\u201C' },
  { tag: '{rdquo}', replacement: '\u201D' },
  { tag: '{it}', replacement: '<i>' },
  { tag: '{/it}', replacement: '</i>' },
  { tag: '{wi}', replacement: '<i>' },
  { tag: '{/wi}', replacement: '</i>' },
  { tag: '{gloss}=', replacement: '<i>' },
  { tag: '{/gloss}', replacement: '</i>' },
  { tag: '{sc}', replacement: '<i>' },
  { tag: '{/sc}', replacement: '</i>' },
  { tag: '{parahw}', replacement: '</i>' },
  { tag: '{/parahw}', replacement: '</i>' },
  { tag: '{phrase}', replacement: '</i>' },
  { tag: '{/phrase}', replacement: '</i>' },
  { tag: '{qword}', replacement: '</i>' },
  { tag: '{/qword}', replacement: '</i>' },
  { tag: '{itsc}', replacement: '<i>' },
  { tag: '{/itsc}', replacement: '</i>' },
  { tag: '{inf}', replacement: '' },
  { tag: '{/inf}', replacement: '' },
  { tag: '{sup}', replacement: '' },
  { tag: '{/sup}', replacement: '' },
  { tag: '{bc}', replacement: '' },
  { tag: '{dx}', replacement: '' },
  { tag: '{/dx}', replacement: '' },
  { tag: '{dx_def}', replacement: '' },
  { tag: '{/dx_def}', replacement: '' },
  { tag: '{dx_ety}', replacement: '' },
  { tag: '{/dx_ety}', replacement: '' },
  { tag: '{ma}', replacement: '' },
  { tag: '{/ma}', replacement: '' },
  { tag: '{rom}', replacement: '' },
  { tag: '{/rom}', replacement: '' }
];

export const COMPLEX_TAGS: ComplexTagType[] = [
  { opening: '{sx', closing: '}' },
  { opening: '{d_link', closing: '}' },
  { opening: '{a_link', closing: '}' },
  { opening: '{i_link', closing: '}' },
  { opening: '{et_link', closing: '}' },
  { opening: '{mat', closing: '}' },
  { opening: '{dxt', closing: '}' }
];

export const COMPLEX_TAGS_TO_REMOVE: ComplexTagType[] = [
  { opening: '{ds|', closing: '}' }
];

export const HW_TAGS: TagType[] = [
  { tag: '{bit}', replacement: '' },
  { tag: '{/bit}', replacement: '' }
];

// Uncomment more tenses as needed
export const TENSES: Record<Tense, string> = {
  pind: 'Presente indicativo',
  pprf: 'Pretérito perfecto simple',
  impf: 'Imperativo',
  pret: 'Pretérito imperfecto'
  //   futr: 'Futuro indicativo',
  //   cond: 'Condicional',
  //   psub: 'Presente subjentivo',
  //   pisb1: 'Pretérito imperfecto',
  //   pisb2: 'Pretérito imperfecto 2',
  //   fsub: 'Futuro',
  //   ppci: 'Pretérito perfecto compuesto',
  //   ppsi: 'Pretérito pluscuamperfecto',
  //   pant: 'Pretérito anterior',
  //   fpin: 'Futuro perfecto',
  //   cpef: 'Condicional perfecto',
  //   ppfs: 'Pretérito perfecto',
  //   ppss1: 'Pretérito pluscuamperfecto',
  //   ppss2: 'Pretérito pluscuamperfecto 2',
  //   fpsb: 'Futuro perfecto',
};
