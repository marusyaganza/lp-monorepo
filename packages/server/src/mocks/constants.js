export const mockEnv = {
  SALT_NUMBER: 3,
  JWT_SECTET: 'JWT_secret'
};
