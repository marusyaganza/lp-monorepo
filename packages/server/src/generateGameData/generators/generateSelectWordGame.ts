import { uniq } from 'lodash';
import { DEFAULT_LANGUAGE } from '../../constants/defaultValues';
import { WordModel } from '../../db/models/Word/Word';
import {
  Game,
  GameData,
  GameDataInput,
  GameSettings,
  Word
} from '../../generated/graphql';
import {
  generateOptions,
  getExamples,
  getGameTask,
  prepareDef
} from '../helpers';

export async function generateSelectWordGame(
  words: Word[],
  parameters: GameDataInput,
  config: GameSettings,
  user: string
): Promise<GameData> {
  const gameType = Game.SelectWord;
  const language = parameters.language || DEFAULT_LANGUAGE;
  const optionsCandidates = await WordModel.selectWordsForOptions(
    gameType,
    language,
    config,
    user
  );

  const questions = words.map(word => {
    const { name, shortDef, id, audioUrl, defs, imgUrl } = word;
    const opts = generateOptions(
      optionsCandidates,
      config.optionsPerGame,
      word,
      language
    );
    const options: string[] = opts.map(opt => opt.name);
    const answer = [name];
    const examples = getExamples(defs);
    const question = uniq(
      shortDef.slice(0, 3).map(def => prepareDef(def, name))
    );

    return {
      answer,
      question,
      wordId: id,
      options,
      additionalInfo: {
        audioUrl,
        examples,
        imgUrl
      }
    };
  });

  const task = getGameTask(gameType);

  return { questions, task, type: gameType };
}
