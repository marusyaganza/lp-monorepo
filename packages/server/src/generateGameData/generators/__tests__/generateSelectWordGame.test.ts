import { testData } from '../../../tests/mocks/wordsForGames';
import { generateSelectWordGame } from '../generateSelectWordGame';
import { Game, Language, Word } from '../../../generated/graphql';
import {
  connectToDb,
  disconnectFromDb,
  dropDb,
  seedDb
} from '../../../tests/helpers/testDBUtils';
import { newWordInputs } from '../../../tests/mocks/inputs/newWordInputs';
import { usersTestData } from '../../../tests/mocks/inputs/newUserInput';
import { DEFAULT_GAMES_SETTINGS } from '../../../constants/defultGameSettings';

const gameType = Game.SelectWord;
const gameConfig = DEFAULT_GAMES_SETTINGS[gameType];

const languages = Object.values(Language);
const task = 'Select a word that means ';
describe('generateSelectWordGame', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  beforeEach(async () => {
    await connectToDb();
    await dropDb();
  });

  afterEach(async () => {
    await dropDb();
    await disconnectFromDb();
  });

  languages.forEach(language => {
    it(`should return the correct number of questions when data has enough ${language} words`, async () => {
      const data = await seedDb({
        words: [
          ...newWordInputs[Language.Spanish],
          ...newWordInputs[Language.English]
        ],
        users: [usersTestData[0]]
      });
      const wordId = data?.words?.[0] as string;
      const userId = data?.users?.[0] as string;
      const words = [...testData[language]] as Word[];
      words[0].id = wordId;
      const gameData = await generateSelectWordGame(
        words,
        {
          language,
          gameType
        },
        gameConfig,
        userId
      );
      expect(
        gameData.questions?.map(q => ({ ...q, wordId: expect.any(String) }))
      ).toMatchSnapshot();
      expect(gameData?.task).toEqual(task);
      expect(gameData.type).toEqual(gameType);
      expect(gameData?.questions?.length).toBe(gameConfig?.wordsPerGame);
      expect(gameData.type).toBe(gameType);
    });

    it(`generate data with minimal ${language} words`, async () => {
      const data = await seedDb({
        words: [
          newWordInputs[Language.Spanish][0],
          newWordInputs[Language.English][0]
        ],
        users: [usersTestData[0]]
      });
      const wordId = data?.words?.[0] as string;
      const userId = data?.users?.[0] as string;
      const words = [testData[language][0]] as Word[];
      words[0].id = wordId;
      const gameData = await generateSelectWordGame(
        words,
        {
          language: language,
          gameType
        },
        gameConfig,
        userId
      );

      expect(gameData?.questions?.length).toBe(gameConfig?.minWords);
      expect(gameData.type).toBe(gameType);
      expect(gameData?.questions?.[0]).toMatchSnapshot({
        wordId: expect.any(String)
      });
    });
  });
});
