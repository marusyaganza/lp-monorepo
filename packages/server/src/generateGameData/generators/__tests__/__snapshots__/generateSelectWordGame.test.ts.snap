// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSelectWordGame generate data with minimal ENGLISH words 1`] = `
{
  "additionalInfo": {
    "audioUrl": "https://example.com/audio/apple.mp3",
    "examples": [],
    "imgUrl": "https://example.com/images/apple.jpg",
  },
  "answer": [
    "apple",
  ],
  "options": [
    "serendipity",
    "quix",
    "flibberfazzle",
    "zephral",
    "glitzephyr",
    "apple",
  ],
  "question": [
    "Round fruit with red or green skin.",
  ],
  "wordId": Any<String>,
}
`;

exports[`generateSelectWordGame generate data with minimal SPANISH words 1`] = `
{
  "additionalInfo": {
    "audioUrl": "https://example.com/audio/manzana.mp3",
    "examples": [
      {
        "text": "Ella comió una manzana roja para el desayuno.",
        "translation": "She ate a red apple for breakfast.",
      },
    ],
    "imgUrl": "https://example.com/images/manzana.jpg",
  },
  "answer": [
    "manzana",
  ],
  "options": [
    "espirelar",
    "trutamor",
    "sonrisanto",
    "danzaroma",
    "aventuraíz",
    "manzana",
  ],
  "question": [
    "Round fruit with red or green skin.",
  ],
  "wordId": Any<String>,
}
`;

exports[`generateSelectWordGame should return the correct number of questions when data has enough ENGLISH words 1`] = `
[
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/apple.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/apple.jpg",
    },
    "answer": [
      "apple",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "apple",
    ],
    "question": [
      "Round fruit with red or green skin.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/orange.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/orange.jpg",
    },
    "answer": [
      "orange",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "orange",
    ],
    "question": [
      "Citrus fruit with [...] rind.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/banana.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/banana.jpg",
    },
    "answer": [
      "banana",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "banana",
    ],
    "question": [
      "Curved fruit with yellow skin.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/strawberry.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/strawberry.jpg",
    },
    "answer": [
      "strawberry",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "strawberry",
    ],
    "question": [
      "Soft red fruit with seeds.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/pineapple.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/pineapple.jpg",
    },
    "answer": [
      "pineapple",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "pineapple",
    ],
    "question": [
      "Tropical fruit with spiky rind.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/grape.mp3",
      "examples": [],
      "imgUrl": "https://example.com/images/grape.jpg",
    },
    "answer": [
      "grape",
    ],
    "options": [
      "serendipity",
      "ephemeral",
      "ubiquitous",
      "quixotic",
      "solitude",
      "grape",
    ],
    "question": [
      "Small juicy fruit with smooth skin.",
    ],
    "wordId": Any<String>,
  },
]
`;

exports[`generateSelectWordGame should return the correct number of questions when data has enough SPANISH words 1`] = `
[
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/manzana.mp3",
      "examples": [
        {
          "text": "Ella comió una manzana roja para el desayuno.",
          "translation": "She ate a red apple for breakfast.",
        },
      ],
      "imgUrl": "https://example.com/images/manzana.jpg",
    },
    "answer": [
      "manzana",
    ],
    "options": [
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "hermosura",
      "manzana",
    ],
    "question": [
      "Round fruit with red or green skin.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/naranja.mp3",
      "examples": [
        {
          "text": "Ella bebió jugo de naranja fresco esta mañana.",
          "translation": "She drank fresh orange juice this morning.",
        },
        {
          "text": "Llamaban la naranja “fruta del sol” por su color.",
          "translation": "They called the orange "sun fruit" because of its color.",
        },
      ],
      "imgUrl": "https://example.com/images/naranja.jpg",
    },
    "answer": [
      "naranja",
    ],
    "options": [
      "correr",
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "naranja",
    ],
    "question": [
      "Citrus fruit with orange rind.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/platano.mp3",
      "examples": [
        {
          "text": "Él peló un plátano y se lo comió.",
          "translation": "He peeled a banana and ate it.",
        },
      ],
      "imgUrl": "https://example.com/images/platano.jpg",
    },
    "answer": [
      "plátano",
    ],
    "options": [
      "correr",
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "plátano",
    ],
    "question": [
      "Curved fruit with yellow skin.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/fresa.mp3",
      "examples": [
        {
          "text": "Ella recogió una fresa madura del jardín.",
          "translation": "She picked a ripe strawberry from the garden.",
        },
      ],
      "imgUrl": "https://example.com/images/fresa.jpg",
    },
    "answer": [
      "fresa",
    ],
    "options": [
      "correr",
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "fresa",
    ],
    "question": [
      "Soft red fruit with seeds.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/pina.mp3",
      "examples": [
        {
          "text": "Hicieron un batido de piña.",
          "translation": "They made a pineapple smoothie.",
        },
      ],
      "imgUrl": "https://example.com/images/pina.jpg",
    },
    "answer": [
      "piña",
    ],
    "options": [
      "correr",
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "piña",
    ],
    "question": [
      "Tropical fruit with spiky rind.",
    ],
    "wordId": Any<String>,
  },
  {
    "additionalInfo": {
      "audioUrl": "https://example.com/audio/uva.mp3",
      "examples": [
        {
          "text": "Él me ofreció un racimo de uvas.",
          "translation": "He offered me a bunch of grapes.",
        },
      ],
      "imgUrl": "https://example.com/images/uva.jpg",
    },
    "answer": [
      "uva",
    ],
    "options": [
      "correr",
      "felicidad",
      "rápidamente",
      "grande",
      "pensar",
      "uva",
    ],
    "question": [
      "Small juicy fruit with smooth skin.",
    ],
    "wordId": Any<String>,
  },
]
`;
